hudi-common/src/main/java/org/apache/hudi/common/table/TableSchemaResolver.java

```diff
@@ -493,4 +496,51 @@ public static MessageType readSchemaFromLogFile(FileSystem fs, Path path) throws
     }
     return null;
   }
+
+  /**
+   * Gets the InternalSchema for a hoodie table from the HoodieCommitMetadata of the instant.
+   *
+   * @return InternalSchema for this table
+   */
+  public Option<InternalSchema> getTableInternalSchemaFromCommitMetadata() {
+    HoodieTimeline timeline = metaClient.getActiveTimeline().getCommitsTimeline().filterCompletedInstants();
```

Comment:
Instead of reading internal metadata from commit file, can we  read it from the .hoodie/.schema folder (using FileBaseInternalSchemasManager).

Reply:
I feel OK。 Just a little performance  worried，  as historySchema will gradually increase,   read from commit file has better performance than read from FileBaseInternalSchemasManager.

Reply:
Save it as an indexed file (HFile) so you can just read the last record or first record and be done? Having one source of truth would be good. We can also do this as follow up